src/Ganeti/HTools/Loader.hs

```diff
@@ -388,11 +388,10 @@ setStaticKvmNodeMem nl static_node_mem =
 -- | Update node memory stat based on instance list.
 updateMemStat :: Node.Node -> Instance.List -> Node.Node
 updateMemStat node il =
-    let missing = truncate (Node.tMem node)
-                  - Node.nMem node
-                  - Node.fMem node
-                  - nodeImem node il
-    in node { Node.xMem = missing }
+  let node2 = node { Node.iMem = nodeImem node il }
```

Comment:
We don't have lens support for these objects, but I think we could add it. I'm not entirely sure if they would help here.

The problem here is that all those functions (nodeImem, missingMem, unallocatedMem) take a node as an argument but they also depend on each other - i.e. missingMem expects that iMem was updated, unallocatedMem expects xMem to be updated beforehand. Maybe I should have defined those functions taking multiple integer arguments instead?